addons/website_forum/models/forum.py

```diff
@@ -982,10 +982,10 @@ def write(self, values):
                     upvote = True
                 elif (values['vote'] == '-1' or vote.vote == '1' and values['vote'] == '0'):
                     upvote = False
-                self._check_karma_rights(upvote)
```

Comment:
there is a `_check_general_rights` above that should either be outside of the loop or use `vote` as well

Reply:
Indeed, I mistook it for a `check_access_rights` which wouldn't be useful in the loop, but wouldn't be wrong either.